.github/workflows/unit_tests.yaml

```diff
@@ -199,6 +199,11 @@ jobs:
     runs-on: ubuntu-latest
     container: spack/github-actions:centos6
     steps:
+    - name: Setup git configuration
+      run: |
+        # Need this for the git tests to succeed.
+        git --version
+        . .github/workflows/setup_git.sh
```

Comment:
Here we are trying to source a file before cloning the Spack repository. I think this line needs to be added twice in each branch of the test (with and without coverage). 

It's unfortunate, but `centos6` is so old that we can't use the `actions/checkout` action inside it if I recall correctly, so we resort to cloning Spack manually.